File: src/main/java/com/cpw/browser/editor/BrowserFileEditorProvider.java
package com.cpw.browser.editor;

import com.intellij.openapi.fileEditor.FileEditor;
import com.intellij.openapi.fileEditor.FileEditorPolicy;
import com.intellij.openapi.fileEditor.FileEditorProvider;
import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

// 浏览器面板的 FileEditorProvider，支持在编辑区显示浏览器
public class BrowserFileEditorProvider implements FileEditorProvider, DumbAware {

    @Override
    public boolean accept(@NotNull Project project, @NotNull VirtualFile file) {
        // 仅接受浏览器文件类型
        return file.getFileType() == BrowserFileType.INSTANCE;
    }

    @Override
    public @NotNull FileEditor createEditor(@NotNull Project project, @NotNull VirtualFile file) {
        return new BrowserFileEditor(project, file);
    }

    @Override
    public @NotNull String getEditorTypeId() {
        return "webbrowser-editor";
    }

    @Override
    public @NotNull FileEditorPolicy getPolicy() {
        return FileEditorPolicy.HIDE_DEFAULT_EDITOR;
    }
}

File: src/main/kotlin/com/cpw/browser/editor/BrowserFileEditorProvider.kt
package com.cpw.browser.editor

import com.intellij.openapi.fileEditor.FileEditor
import com.intellij.openapi.fileEditor.FileEditorPolicy
import com.intellij.openapi.fileEditor.FileEditorProvider
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

/**
 * 浏览器面板的 FileEditorProvider，支持在编辑区显示浏览器。
 */
class BrowserFileEditorProvider : FileEditorProvider, DumbAware {

    override fun accept(project: Project, file: VirtualFile): Boolean {
        return file.fileType == BrowserFileType
    }

    override fun createEditor(project: Project, file: VirtualFile): FileEditor {
        return BrowserFileEditor(project, file)
    }

    override fun getEditorTypeId(): String = "webbrowser-editor"
    override fun getPolicy(): FileEditorPolicy = FileEditorPolicy.HIDE_DEFAULT_EDITOR
}
